lib/puppet/provider/zpool/zpool.rb

```diff
@@ -34,7 +34,17 @@ def process_zpool_data(pool_array)
         sym = (value =~ %r{^mirror}) ? :mirror : :raidz
         pool[:raid_parity] = 'raidz2' if value =~ %r{^raidz2}
       else
-        tmp << value
+        # get full drive name if the value is a partition (Linux only)
+        if Facter.value(:kernel) == 'Linux'
+          disks = if value =~ %r{/dev/(:?[a-z]+1|disk/by-id/.+-part1)$}
+                    execute("lsblk -p -no pkname #{value}").chomp
+                  else
+                    value
+                  end
+          tmp << disks
+        else
+          tmp << value
+        end
```

Comment:
I think this can be written as:
```ruby
if Facter.value(:kernel) == 'Linux' && value =~ %r{/dev/(:?[a-z]+1|disk/by-id/.+-part1)$}
  tmp << execute("lsblk -p -no pkname #{value}").chomp
else
  tmp << value
end
```